src/Microsoft.Framework.Runtime/DependencyManagement/ProjectReferenceDependencyProvider.cs

```diff
@@ -45,7 +45,7 @@ public LibraryDescription GetDescription(LibraryRange libraryRange, FrameworkNam
 
             // This never returns null
             var targetFrameworkInfo = project.GetTargetFramework(targetFramework);
-            var targetFrameworkDependencies = targetFrameworkInfo.Dependencies;
+            var targetFrameworkDependencies = new List<LibraryDependency>(targetFrameworkInfo.Dependencies);
```

Comment:
It means we need to clone a `TargetFrameworkInformation` instance each time `Project.GetTargetFramework()` is called.

To be consistent, we even need to clone all frameworks each time `Project.GetTargetFrameworks()` is called.

Too expensive. Do you still want it?